src/AppInstallerCommonCore/PathVariable.cpp

```diff
@@ -81,4 +107,16 @@ namespace AppInstaller::Registry::Environment
         std::wstring pathName = std::wstring{ s_PathName };
         m_key.SetValue(pathName, ConvertToUTF16(value), REG_EXPAND_SZ);
     }
+
+    bool RefreshPathVariableForCurrentProcess()
+    {
+        std::string userPathValue = PathVariable(Manifest::ScopeEnum::User, true).GetPathValue();
+        std::string systemPathValue = PathVariable(Manifest::ScopeEnum::Machine, true).GetPathValue();
+
+        EnsurePathValueEndsWithSemicolon(userPathValue);
+        EnsurePathValueEndsWithSemicolon(systemPathValue);
+
+        std::wstring pathValue = ConvertToUTF16(userPathValue + systemPathValue);

```

Comment:
per offline discussion, we need to expand the paths before setting them
Also per https://web.archive.org/web/20111225045158/http://support.microsoft.com/kb/100843/EN-US
It should be system path and append with user path

Reply:
I added a new split function to split the path entries with the ';' character, then expanded each path entry and concatenated them together with the ';' character. Also verified that this behavior is necessary by modifying the path with %USERPROFILE%/testPath and executing a batch script in that path within the same refreshed process environment.

Reply:
Changed so that systempath comes first and is appended with user path.